clang/runtime/dpct-rt/include/dpct/math.hpp

```diff
@@ -2216,6 +2218,103 @@ void ldmatrix(uintptr_t addr, T *m1, T *m2, T *m3, T *m4, bool trans = false) {
   ldmatrix(addr, m4, trans, 3);
 }
 
+/// Multiplies 2 matrices (A & B) and adds the result to C matrix and
+/// accumulates the result to a D matrix (MAD). Requires the sub-group size of
+/// kernel calling this function to be 32.
+/// Current supported shapes & types:
+/// - m16n8k16 (f32.f16.f16.f32 & s32.s8.s8.s32)
```

Comment:
For f32.f16.f16.f32 & s32.s8.s8.s32 supported in this helper function, pls write E2E test case to verify the functionality of these two shape for this two types.

Reply:
E2E cases for above are verified